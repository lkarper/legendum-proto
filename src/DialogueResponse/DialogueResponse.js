import React from 'react';
import PropTypes from 'prop-types';
import './DialogueResponse.css';

const DialogueResponse = (props) => {
    const { 
        dialogue, 
        choiceIndex, 
        page,
    } = props;

    if (dialogue.length === 0) {
        return (
            <p 
                className='DialogueResponse__text'
            >
                Error: Something went wrong. Check your connection and try again.
            </p>
        );
    } else {
        const { responses_to_choices } = dialogue[page - 1];

        const responsesArray = responses_to_choices.split("|");
        
        return (
            <p 
                className='DialogueResponse__text'
            >
                {responsesArray[choiceIndex]}
            </p>
        );
    }
}

DialogueResponse.defaultProps = { 
    dialogue: [], 
    choiceIndex: 0, 
    page: 1,
};

DialogueResponse.propTypes = { 
    dialogue: PropTypes.array, 
    choiceIndex: PropTypes.number, 
    page: PropTypes.number,
};

export default DialogueResponse;
