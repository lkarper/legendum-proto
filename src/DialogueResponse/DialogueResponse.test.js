import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DialogueResponse from './DialogueResponse';

describe('DialogueResponse component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DialogueResponse />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<DialogueResponse />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
