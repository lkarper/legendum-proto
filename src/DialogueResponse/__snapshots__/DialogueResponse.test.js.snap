// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogueResponse component renders the UI as expected 1`] = `
<DialogueResponse
  choiceIndex={0}
  dialogue={Array []}
  page={1}
>
  <p
    className="DialogueResponse__text"
  >
    Error: Something went wrong. Check your connection and try again.
  </p>
</DialogueResponse>
`;
